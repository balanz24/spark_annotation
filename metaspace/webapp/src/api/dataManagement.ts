import gql from 'graphql-tag'

export interface GroupListItem {
  id: string;
  name: string;
}

export const oneGroupQuery =
  gql`query oneGroupQuery($groupId: ID!) {
  group(groupId:$groupId) {
    id
    name
  }
}`

export const oneProjectQuery =
  gql`query oneProjectQuery($projectId: ID!) {
  project(projectId:$projectId) {
    id
    name
  }
}`

export const allGroupsQuery =
gql`query allGroupsQuery($query: String) {
  allGroups(query:$query) {
    id
    name
  }
}`

export const requestAccessToGroupMutation =
gql`mutation($groupId: ID!) {
  requestAccessToGroup(groupId: $groupId){
    role
    }
  }`
