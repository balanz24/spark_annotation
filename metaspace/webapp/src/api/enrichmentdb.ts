import gql from 'graphql-tag'

export interface EnrichmentDB {
  id: number
  name: string
}

export interface EnrichmentTerm {
  id: number
  enrichmentId: string
  enrichmentName: string
  enrichmentDB: EnrichmentDB
}

export const checkIfEnrichmentRequested =
gql`query enrichmentRequested($id: String!) {
  enrichmentRequested(datasetId: $id)
}`

export const getEnrichedMolDatabasesQuery =
gql`query allEnrichedMolDatabases($id: String!) {
  allEnrichedMolDatabases(datasetId: $id) {
    id
    name
    version
    archived
    group {
      id
      shortName
    }
  }
}`
