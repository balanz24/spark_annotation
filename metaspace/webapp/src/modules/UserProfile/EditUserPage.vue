<template>
  <div class="main-content">
    <el-dialog
      title="Delete account"
      custom-class="delete-account-dialog"
      :visible.sync="showDeleteAccountDialog"
      width="500px"
      :lock-scroll="false"
    >
      <p>
        If you delete your account, you will lose access to all private datasets.
      </p>
      <p>
        Please select whether you would like to delete all your datasets or keep them within METASPACE accessible by the group members:
      </p>
      <!--Changed the Text below to the above version after discussion with Theo, plz consider if it fits-->
      <!--If you delete your account, you will lose access to any datasets, groups and-->
      <!--projects that have been explicitly shared with you-->
      <el-checkbox
        v-model="delDatasets"
        style="margin: 10px 20px"
      >
        Delete datasets that I have submitted
      </el-checkbox>
      <div style="margin: 10px 0 20px; text-align: right;">
        <el-button
          title="Cancel"
          @click="closeDeleteAccountDialog"
        >
          Cancel
        </el-button>
        <el-button
          type="danger"
          title="Delete account"
          :loading="isUserDeletionLoading"
          @click="deleteAccount()"
        >
          Delete account
        </el-button>
      </div>
      <p>
        <b>Note:</b> if you choose not to delete the datasets now, you will still be able to have them
        deleted later by emailing the <a href="mailto:contact@metaspace2020.eu">METASPACE administrators</a>.
      </p>
    </el-dialog>
    <div
      v-loading="!isLoaded"
      class="user-edit-page"
    >
      <el-row>
        <el-col :span="16">
          <h2>User details</h2>
        </el-col>
        <el-col :span="8">
          <el-button
            title="Save"
            type="primary"
            :disabled="isUserDetailsPristine"
            class="saveButton"
            :loading="isUserDetailsLoading"
            @click="updateUserDetails"
          >
            Save
          </el-button>
        </el-col>
      </el-row>
      <el-row :gutter="20">
        <el-form
          ref="form"
          :disabled="isUserDetailsLoading"
          :rules="rules"
          :model="model"
          label-position="top"
        >
          <div>
            <el-col :span="8">
              <el-form-item
                prop="name"
                label="Full name"
              >
                <el-input
                  v-model="model.name"
                  name="name"
                />
              </el-form-item>
            </el-col>
            <el-col :span="8">
              <el-form-item
                prop="email"
                label="Email address"
              >
                <el-input
                  v-model="model.email"
                  name="email"
                />
              </el-form-item>
              <p v-if="isEmailChangePending">
                <b>Please click the link that has been sent to your new email address to verify the change.</b>
              </p>
            </el-col>
            <el-col :span="8">
              <el-form-item
                label="Password"
                required
              >
                <el-button
                  style="margin-left: 16px"
                  type="text"
                  @click="handleChangePassword"
                >
                  Click to change...
                </el-button>
              </el-form-item>
            </el-col>
          </div>
        </el-form>
      </el-row>

      <div>
        <h2>Groups</h2>
        <groups-table
          :current-user="currentUser"
          :refetch-data="refetchData"
        />
        <div v-if="currentUser && currentUser.groups && currentUser.groups.length > 1">
          <p>Primary group:</p>
          <el-select
            v-model="primaryGroupId"
            v-loading="isChangingPrimaryGroup"
            placeholder="Select"
            style="width: 400px;"
            @change="handleChangePrimaryGroup"
          >
            <el-option
              v-for="userGroup in currentUser.groups"
              :key="userGroup.group.id"
              :label="userGroup.group.name"
              :value="userGroup.group.id"
            />
          </el-select>
        </div>
      </div>

      <div style="margin-top: 40px;">
        <h2>Projects</h2>
        <projects-table
          :current-user="currentUser"
          :refetch-data="refetchData"
        />
      </div>
      <!--The section below will be introduced in vFuture-->
      <!--<div class="notifications" style="margin-top: 30px">-->
      <!--<h2>Notifications</h2>-->
      <!--<div class="notification-list" style="padding-left: 10px">-->
      <!--<el-row :gutter="0">-->
      <!--<el-col :span="12"><p>Send an email when:</p>-->
      <!--<el-checkbox-group-->
      <!--class="notifications">-->
      <!--<div class="notifications_checkbox">-->
      <!--<el-checkbox-->
      <!--label="My dataset's processing has successfully finished"></el-checkbox>-->
      <!--</div>-->
      <!--<div class="notifications_checkbox">-->
      <!--<el-checkbox class="notifications_checkbox"-->
      <!--label="My dataset's processing failed"></el-checkbox>-->
      <!--</div>-->
      <!--</el-checkbox-group>-->
      <!--<p>Show a notification when:</p>-->
      <!--<el-checkbox-group :indeterminate="isIndeterminate" v-model="checkList"-->
      <!--class="notifications">-->
      <!--<div class="notifications_checkbox">-->
      <!--<el-checkbox class="notifications_checkbox"-->
      <!--label="My dataset's processing has successfully finished"></el-checkbox>-->
      <!--</div>-->
      <!--<div class="notifications_checkbox">-->
      <!--<el-checkbox class="notifications_checkbox"-->
      <!--label="A dataset is added to a group or project that I belong to"></el-checkbox>-->
      <!--</div>-->
      <!--<div class="notifications_checkbox">-->
      <!--<el-checkbox class="notifications_checkbox"-->
      <!--label="A public dataset has been added to the queue"></el-checkbox>-->
      <!--</div>-->
      <!--<div class="notifications_checkbox">-->
      <!--<el-checkbox class="notifications_checkbox"-->
      <!--label="A public dataset has finished processing"></el-checkbox>-->
      <!--</div>-->
      <!--</el-checkbox-group>-->
      <!--</el-col>-->
      <!--</el-row>-->
      <!--</div>-->
      <!--</div>-->
      <section>
        <h2>API access</h2>
        <div class="action-with-message">
          <p class="max-w-measure-4">
            To <a href="https://github.com/metaspace2020/metaspace/tree/master/metaspace/python-client">access METASPACE programmatically</a>
            or integrate with trusted third-party applications, an API key can be used to avoid sharing your password.
          </p>
          <el-button
            v-if="currentUser && !currentUser.apiKey"
            @click="handleGenerateApiKey"
          >
            Generate API key
          </el-button>
        </div>
        <el-row v-if="isLoaded && currentUser">
          <div
            v-if="currentUser.apiKey"
            class="action-with-message"
          >
            <copy-to-clipboard
              class="max-w-measure-1"
              :value="currentUser.apiKey"
              type="password"
            />
            <el-button
              type="danger"
              @click="handleRevokeApiKey"
            >
              Revoke key
            </el-button>
          </div>
        </el-row>
      </section>
      <section>
        <h2>Delete account</h2>
        <div class="action-with-message">
          <p class="max-w-measure-3">
            If you delete your METASPACE account, you can optionally remove all of your datasets.
            If datasets are not removed, the private data will still be accessible by the group members only.
          </p>
          <el-button
            type="danger"
            title="Delete account"
            @click="openDeleteAccountDialog()"
          >
            Delete account
          </el-button>
        </div>
      </section>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { Component, Watch } from 'vue-property-decorator'
import {
  updateUserMutation,
  deleteUserMutation,
  userProfileQuery,
  UserProfileQuery, resetUserApiKeyMutation,
} from '../../api/user'
import reportError from '../../lib/reportError'
import { refreshLoginStatus } from '../../api/graphqlClient'
import { ElForm } from 'element-ui/types/form'
import { TransferDatasetsDialog } from '../Group/index'
import emailRegex from '../../lib/emailRegex'
import GroupsTable from './GroupsTable'
import ProjectsTable from './ProjectsTable.vue'
import ConfirmAsync from '../../components/ConfirmAsync'
import { sendPasswordResetToken } from '../../api/auth'
import CopyToClipboard from '../../components/Form/CopyToClipboard'

  interface Model {
    name: string;
    email: string | null;
  }

  @Component<EditUserPage>({
    components: {
      TransferDatasetsDialog,
      GroupsTable,
      ProjectsTable,
      CopyToClipboard,
    },
    apollo: {
      currentUser: {
        query: userProfileQuery,
        result({ data }: {data: {currentUser: UserProfileQuery}}) {
          if (!this.isLoaded) {
            // Not using 'loadingKey' pattern here to avoid getting a full-page loading spinner when the user clicks a
            // button that causes this query to refetch
            this.isLoaded = true
          }
        },
      },
    },
  })
export default class EditUserPage extends Vue {
    isLoaded = false;
    showApiKey = false;
    showDeleteAccountDialog: boolean = false;
    isUserDetailsLoading: boolean = false;
    isUserDeletionLoading: boolean = false;
    isEmailChangePending: boolean = false;
    isChangingPrimaryGroup: boolean = false;

    currentUser: UserProfileQuery | null = null;
    model: Model = {
      name: '',
      email: '',
    };

    primaryGroupId: string | null = null;

    delDatasets: boolean = false;
    rules: object = {
      name: [
        { required: true, min: 3, max: 50, message: 'Please enter a correct fullname', trigger: 'blur' },
      ],
      email: [
        {
          required: true,
          pattern: emailRegex,
          message: 'Please enter a valid email address',
          trigger: 'blur',
        }],
    };

    get isUserDetailsPristine() {
      return this.currentUser == null
        || (this.model.name === this.currentUser.name && this.model.email === this.currentUser.email)
    }

    @Watch('isLoaded')
    @Watch('currentUser', { deep: true })
    onCurrentUserChanged(this: any) {
      if (this.currentUser) {
        this.model.name = this.currentUser.name
        if (!this.isEmailChangePending) {
          this.model.email = this.currentUser.email
        }
        this.primaryGroupId = this.currentUser.primaryGroup ? this.currentUser.primaryGroup.group.id : null
      } else if (this.isLoaded) {
        this.$router.push('/account/sign-in')
      }
    }

    openDeleteAccountDialog() {
      this.showDeleteAccountDialog = true
    }

    closeDeleteAccountDialog() {
      this.showDeleteAccountDialog = false
    }

    async updateUserDetails() {
      try {
        await (this.$refs.form as ElForm).validate()
      } catch (err) {
        return
      }
      try {
        const emailChanged = this.currentUser!.email !== this.model.email
        if (emailChanged) {
          try {
            await this.$confirm(
              'Are you sure you want to change email address? '
              + 'A verification email will be sent to your new address to confirm the change.',
              'Confirm email address change', {
                confirmButtonText: 'Yes, send verification email',
                lockScroll: false,
              })
            this.isEmailChangePending = true
          } catch {
            return
          }
        }
        this.isUserDetailsLoading = true

        await this.$apollo.mutate({
          mutation: updateUserMutation,
          variables: {
            userId: this.currentUser!.id,
            update: {
              // Only send fields that have changed, per API requirements
              // This relies on `undefined` values being discarded during JSON stringification
              name: this.model.name !== this.currentUser!.name ? this.model.name : undefined,
              email: this.model.email !== this.currentUser!.email ? this.model.email : undefined,
            },
          },
        })
        await this.$apollo.queries.currentUser.refetch()
        this.$message({
          type: 'success',
          message: emailChanged
            ? 'A verification link has been sent to your new email address. '
            + 'Please click the link in this email to confirm the change.'
            : 'Your details have been saved',
        })
      } catch (err) {
        reportError(err)
      } finally {
        this.isUserDetailsLoading = false
      }
    }

    async deleteAccount(this: any) {
      try {
        this.isUserDeletionLoading = true
        await this.$apollo.mutate({
          mutation: deleteUserMutation,
          variables: {
            userId: this.currentUser.id,
            deleteDatasets: this.delDatasets,
          },
        })
        this.$message({
          type: 'success',
          message: 'You have successfully deleted your account!',
        })
      } catch (err) {
        reportError(err)
      } finally {
        await refreshLoginStatus()
        this.closeDeleteAccountDialog()
        this.isUserDeletionLoading = false
      }
    }

    @ConfirmAsync(function(this: EditUserPage) {
      return {
        message: 'This will send you an email with a link and instructions to change your password. '
          + 'Do you wish to proceed?',
        confirmButtonText: 'Change password',
        confirmButtonLoadingText: 'Sending email...',
      }
    })
    async handleChangePassword() {
      // TODO: Customize this so it's not so obviously a rip off of the reset password process
      await sendPasswordResetToken(this.currentUser!.email!)
      this.$message({ message: 'Email sent!', type: 'success' })
    }

    async handleChangePrimaryGroup() {
      const oldPrimaryGroupId = this.currentUser!.primaryGroup != null ? this.currentUser!.primaryGroup!.group.id : null

      if (this.primaryGroupId !== oldPrimaryGroupId) {
        this.isChangingPrimaryGroup = true
        try {
          await this.$apollo.mutate({
            mutation: updateUserMutation,
            variables: {
              userId: this.currentUser!.id,
              update: { primaryGroupId: this.primaryGroupId },
            },
          })
          await this.$apollo.queries.currentUser.refetch()
        } finally {
          this.isChangingPrimaryGroup = false
        }
      }
    }

    @ConfirmAsync(function(this: EditUserPage) {
      return {
        dangerouslyUseHTMLString: true,
        message: `API keys allow almost full access to your account, including data shared with you by others,
        and should not be shared with anybody you don't trust.<br><br>
        If you are no longer using your API key or you accidentally share it publicly,
        revoke the key to prevent unwanted access to your account.`,
        confirmButtonText: 'I understand, generate the key',
        confirmButtonLoadingText: 'Generating...',
      }
    })
    async handleGenerateApiKey() {
      await this.$apollo.mutate({
        mutation: resetUserApiKeyMutation,
        variables: {
          userId: this.currentUser!.id,
          removeKey: false,
        },
      })
      await this.$apollo.queries.currentUser.refetch()
    }

    @ConfirmAsync(function(this: EditUserPage) {
      return {
        message: 'Are you sure you want to revoke the API key?',
        confirmButtonText: 'Revoke key',
        confirmButtonLoadingText: 'Revoking...',
      }
    })
    async handleRevokeApiKey() {
      await this.$apollo.mutate({
        mutation: resetUserApiKeyMutation,
        variables: {
          userId: this.currentUser!.id,
          removeKey: true,
        },
      })
      await this.$apollo.queries.currentUser.refetch()
    }

    handleCopyApiKey() {
      if (this.currentUser && this.currentUser.apiKey) {
        if ('clipboard' in navigator) {
          navigator.clipboard.writeText(this.currentUser.apiKey)
        } else {
          const el = document.createElement('textarea')
          el.value = this.currentUser!.apiKey
          el.style.position = 'absolute'
          el.style.left = '-9999px'
          document.body.appendChild(el)
          try {
            el.select()
            document.execCommand('copy')
          } finally {
            document.body.removeChild(el)
          }
        }
      }
    }

    async refetchData() {
      await this.$apollo.queries.currentUser.refetch()
    }
}
</script>

<style scoped>
  .main-content {
    @apply px-6 py-3;
    display: flex;
    justify-content: center;
  }

  .user-edit-page {
    max-width: 950px;
    width: 100%;
  }

  .api-key /deep/ input {
    background-color: white !important;
    cursor: text !important;
  }

  .saveButton {
    width: 100px;
    padding: 8px;
    float: right;
    margin: 20px 0;
  }

  /deep/ .delete-account-dialog .el-dialog__body {
    padding: 0 20px 20px 20px;
  }

  /* Uncomment when the vFuture notifications will be introduced */
  /*.notifications_checkbox {*/
    /*margin-left: 0;*/
    /*padding: 0;*/
  /*}*/

  .action-with-message {
    align-items: flex-start;
    display: flex;
    justify-content: space-between;
  }

  .action-with-message > :first-child {
    margin-top: 0;
    margin-right: 20px;
  }

  .action-with-message p {
    font-size: 16px;
    line-height: 1.5;
  }

  section + section {
    @apply my-12;
  }
</style>
