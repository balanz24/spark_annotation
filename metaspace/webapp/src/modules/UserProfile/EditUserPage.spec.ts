import { mount, Wrapper } from '@vue/test-utils'
import Vue from 'vue'
import { DeepPartial } from 'ts-essentials'
import { restoreConsole, suppressConsoleWarn } from '../../../tests/utils/suppressConsole'
import EditUserPage from './EditUserPage.vue'
import router from '../../router'
import { initMockGraphqlClient, apolloProvider } from '../../../tests/utils/mockGraphqlClient'
import { UserProfileQuery } from '../../api/user'

describe('EditUserPage', () => {
  const mockCurrentUser: DeepPartial<UserProfileQuery> = {
    id: '22333',
    name: 'foo',
    email: 'foo@bar.baz',
    role: 'user',
    groups: [
      { role: 'MEMBER', numDatasets: 0, group: { id: 'AAA', name: 'Group A', urlSlug: 'grp-a' } },
      { role: 'INVITED', numDatasets: 0, group: { id: 'BBB', name: 'Group B', urlSlug: null } },
      { role: 'PENDING', numDatasets: 0, group: { id: 'CCC', name: 'Group C', urlSlug: null } },
      { role: 'GROUP_ADMIN', numDatasets: 20, group: { id: 'DDD', name: 'Group D', urlSlug: null } },
    ],
    primaryGroup: { role: 'GROUP_ADMIN', numDatasets: 20, group: { id: 'DDD', name: 'Group D', urlSlug: null } },
    projects: [
      { role: 'MEMBER', numDatasets: 0, project: { id: 'AA', name: 'Project A', urlSlug: 'proj-a' } },
      { role: 'INVITED', numDatasets: 0, project: { id: 'BB', name: 'Project B', urlSlug: null } },
      { role: 'PENDING', numDatasets: 0, project: { id: 'CC', name: 'Project C', urlSlug: null } },
      { role: 'MANAGER', numDatasets: 20, project: { id: 'DD', name: 'Project D', urlSlug: null } },
    ],
  }

  const mockUpdateUserMutation = jest.fn((src: any, args: any, ctx: any) => ({}))

  beforeEach(() => {
    suppressConsoleWarn('async-validator:')
    jest.clearAllMocks()
    initMockGraphqlClient({
      Query: () => ({
        currentUser: () => mockCurrentUser,
      }),
      Mutation: () => ({
        updateUser: mockUpdateUserMutation,
      }),
    })
  })

  afterEach(async() => {
    restoreConsole()
  })

  it('should match snapshot', async() => {
    const wrapper = mount(EditUserPage, { router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('should be able to submit changes to the user', async() => {
    const wrapper = mount(EditUserPage, { router, apolloProvider })
    await Vue.nextTick()
    const nameInput = wrapper.find('input[name="name"]')
    const emailInput = wrapper.find('input[name="email"]')
    const saveButton = wrapper.find('.saveButton')
    const name = 'foo bar'
    const email = 'changed@bar.baz'
    wrapper.vm.$confirm = jest.fn(() => Promise.resolve('confirm'))
    await Vue.nextTick()

    nameInput.setValue(name)
    emailInput.setValue(email)
    await Vue.nextTick()
    saveButton.trigger('click')
    await Vue.nextTick()

    expect(mockUpdateUserMutation).toHaveBeenCalledTimes(1)
    expect(mockUpdateUserMutation.mock.calls[0][1]).toEqual(
      expect.objectContaining({
        userId: mockCurrentUser.id,
        update: {
          name,
          email,
        },
      }),
    )
  })

  it('should not include unchanged fields in the update payload', async() => {
    const wrapper = mount(EditUserPage, { router, apolloProvider })
    await Vue.nextTick()
    const nameInput = wrapper.find('input[name="name"]')
    const saveButton = wrapper.find('.saveButton')
    const name = 'foo bar'
    wrapper.vm.$confirm = jest.fn(() => Promise.resolve('confirm'))

    nameInput.setValue(name)
    await Vue.nextTick()
    saveButton.trigger('click')
    await Vue.nextTick()

    expect(mockUpdateUserMutation).toHaveBeenCalledTimes(1)
    expect(mockUpdateUserMutation.mock.calls[0][1]).toEqual(
      expect.objectContaining({
        userId: mockCurrentUser.id,
        update: { name },
      }),
    )
  })
})
