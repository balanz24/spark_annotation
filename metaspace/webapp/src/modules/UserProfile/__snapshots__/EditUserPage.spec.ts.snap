// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserPage should match snapshot 1`] = `
<div class="main-content">
  <mock-el-dialog title="Delete account" custom-class="delete-account-dialog" width="500px">
    <p>
      If you delete your account, you will lose access to all private datasets.
    </p>
    <p>
      Please select whether you would like to delete all your datasets or keep them within METASPACE accessible by the group members:
    </p> <label class="el-checkbox" style="margin: 10px 20px;"><span class="el-checkbox__input"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value=""></span><span class="el-checkbox__label">
      Delete datasets that I have submitted
    <!----></span></label>
    <div style="margin: 10px 0px 20px; text-align: right;"><button type="button" class="el-button el-button--default" title="Cancel">
        <!---->
        <!----><span>
        Cancel
      </span></button> <button type="button" class="el-button el-button--danger" title="Delete account">
        <!---->
        <!----><span>
        Delete account
      </span></button></div>
    <p><b>Note:</b> if you choose not to delete the datasets now, you will still be able to have them
      deleted later by emailing the <a href="mailto:contact@metaspace2020.eu">METASPACE administrators</a>.
    </p>
  </mock-el-dialog>
  <div class="user-edit-page" mock-v-loading="false">
    <div class="el-row">
      <div class="el-col el-col-16">
        <h2>User details</h2>
      </div>
      <div class="el-col el-col-8"><button disabled="disabled" type="button" class="el-button saveButton el-button--primary is-disabled" title="Save">
          <!---->
          <!----><span>
          Save
        </span></button></div>
    </div>
    <div class="el-row" style="margin-left: -10px; margin-right: -10px;">
      <form class="el-form el-form--label-top">
        <div>
          <div class="el-col el-col-8" style="padding-left: 10px; padding-right: 10px;">
            <div class="el-form-item is-required"><label for="name" class="el-form-item__label">Full name</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="text" autocomplete="off" name="name" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <transition-stub name="el-zoom-in-top">
                  <!---->
                </transition-stub>
              </div>
            </div>
          </div>
          <div class="el-col el-col-8" style="padding-left: 10px; padding-right: 10px;">
            <div class="el-form-item is-required"><label for="email" class="el-form-item__label">Email address</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="text" autocomplete="off" name="email" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <transition-stub name="el-zoom-in-top">
                  <!---->
                </transition-stub>
              </div>
            </div>
            <!---->
          </div>
          <div class="el-col el-col-8" style="padding-left: 10px; padding-right: 10px;">
            <div class="el-form-item is-required"><label class="el-form-item__label">Password</label>
              <div class="el-form-item__content"><button type="button" class="el-button el-button--text" style="margin-left: 16px;">
                  <!---->
                  <!----><span>
                Click to change...
              </span></button>
                <transition-stub name="el-zoom-in-top">
                  <!---->
                </transition-stub>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div>
      <h2>Groups</h2>
      <div>
        <table class="sm-table sm-table-user-details">
          <tr>
            <th>Group</th>
            <th>Role</th>
            <th>Datasets</th>
            <th></th>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/group/grp-a" class="">Group A</a>
                <mock-el-tooltip content="Group A has a pending membership request." placement="right"><span class="notification"></span></mock-el-tooltip>
              </div>
            </td>
            <td>Member</td>
            <td>0</td>
            <td>
              <div class="sm-table-button-group"><button type="button" class="el-button el-button--default el-button--mini">
                  <!----><i class="el-icon-arrow-right"></i><span>Leave</span></button></div>
            </td>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/group/BBB" class="">Group B</a></div>
            </td>
            <td>Invited</td>
            <td>0</td>
            <td>
              <div class="sm-table-button-group"><button type="button" class="el-button el-button--success el-button--mini">
                  <!----><i class="el-icon-check"></i><span>Accept</span></button><button type="button" class="el-button el-button--default el-button--mini">
                  <!----><i class="el-icon-close"></i><span>Decline</span></button></div>
            </td>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/group/CCC" class="">Group C</a>
                <mock-el-tooltip content="Group C has a pending membership request." placement="right"><span class="notification"></span></mock-el-tooltip>
              </div>
            </td>
            <td>Requesting access</td>
            <td>0</td>
            <td>
              <div class="sm-table-button-group"></div>
            </td>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/group/DDD" class="">Group D</a></div>
            </td>
            <td>Group admin</td>
            <td><a href="/datasets?grp=DDD&amp;subm=22333" class="">20</a></td>
            <td>
              <div class="sm-table-button-group"><button disabled="disabled" type="button" class="el-button el-button--default el-button--mini is-disabled">
                  <!----><i class="el-icon-arrow-right"></i><span>Leave</span></button></div>
            </td>
          </tr>
        </table>
      </div>
      <div>
        <p>Primary group:</p>
        <mock-el-select placeholder="Select" value="DDD" style="width: 400px;" mock-v-loading="false">
          <mock-el-option label="Group A" value="AAA"></mock-el-option>
          <mock-el-option label="Group B" value="BBB"></mock-el-option>
          <mock-el-option label="Group C" value="CCC"></mock-el-option>
          <mock-el-option label="Group D" value="DDD"></mock-el-option>
        </mock-el-select>
      </div>
    </div>
    <div style="margin-top: 40px;">
      <h2>Projects</h2>
      <div class="el-row">
        <mock-el-dialog append-to-body="" title="Create project" class="dialog">
          <div mock-v-loading="0">
            <form class="el-form leading-6 v-rhythm-3 el-form--label-top" size="mini">
              <div><label><span class="text-base font-medium">Title</span>
                  <div class="el-form-item is-required">
                    <!---->
                    <div class="el-form-item__content">
                      <div class="py-1 el-input">
                        <!----><input type="text" autocomplete="off" max-length="50" min-length="2" class="el-input__inner">
                        <!---->
                        <!---->
                        <!---->
                        <!---->
                      </div>
                      <transition-stub name="el-zoom-in-top">
                        <!---->
                      </transition-stub>
                    </div>
                  </div></label>
                <div class="hidden el-input">
                  <!----><input type="text" autocomplete="off" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
              <div><label><span class="text-base font-medium">Privacy</span> <span class="block text-sm text-gray-800">
        Visibility of the project only, does not apply to included datasets
      </span></label>
                <div class="h-10 flex items-center">
                  <div role="switch" class="el-switch"><input type="checkbox" name="" true-value="true" class="el-switch__input"><span class="el-switch__label el-switch__label--left is-active"><!----><span>Private</span></span><span class="el-switch__core" style="width: 40px;"></span><span class="el-switch__label el-switch__label--right"><!----><span aria-hidden="true">Public</span></span></div>
                </div>
                <!---->
              </div>
            </form>
            <div class="mt-6">
              <h4 class="m-0">
                Would you like to include previously submitted datasets?
              </h4>
              <div>
                <div class="select-buttons"><a href="#">Select none</a> <span> | </span> <a href="#">Select all</a></div>
                <div class="dataset-checkbox-list leading-6">
                  <div><label class="el-checkbox flex h-6 items-center"><span class="el-checkbox__input"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value=""></span><span class="el-checkbox__label"><span title="allDatasets.0.name" class="truncate">
          allDatasets.0.name
        </span> <span class="text-gray-700 text-xs tracking-wide pl-1"><mock-elapsed-time date="allDatasets.0.uploadDT"></mock-elapsed-time></span>
                      <!----></span></label></div>
                  <div><label class="el-checkbox flex h-6 items-center"><span class="el-checkbox__input"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value=""></span><span class="el-checkbox__label"><span title="allDatasets.1.name" class="truncate">
          allDatasets.1.name
        </span> <span class="text-gray-700 text-xs tracking-wide pl-1"><mock-elapsed-time date="allDatasets.1.uploadDT"></mock-elapsed-time></span>
                      <!----></span></label></div>
                </div>
              </div>
            </div>
            <div class="button-bar"><button type="button" class="el-button el-button--default">
                <!---->
                <!----><span>
        Cancel
      </span></button> <button autofocus="autofocus" type="button" class="el-button el-button--primary">
                <!---->
                <!----><span>
        Create project
      </span></button></div>
          </div>
        </mock-el-dialog>
        <table class="sm-table sm-table-user-details">
          <tr>
            <th>Project</th>
            <th>Role</th>
            <th>Datasets</th>
            <th></th>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/project/proj-a" class="">
                  Project A
                </a>
                <mock-el-tooltip content="Project A has a pending membership request." placement="right"><span class="notification"></span></mock-el-tooltip>
              </div>
            </td>
            <td>
              Member
            </td>
            <td>
              <!----> <span>0</span></td>
            <td><button type="button" class="el-button el-button--default el-button--mini">
                <!----><i class="el-icon-arrow-right"></i><span>
          Leave
        </span></button>
              <!---->
              <!---->
              <!---->
            </td>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/project/BB" class="">
                  Project B
                </a>
                <!---->
              </div>
            </td>
            <td>
              Invited
            </td>
            <td>
              <!----> <span>0</span></td>
            <td>
              <!---->
              <!----> <button type="button" class="el-button el-button--success el-button--mini">
                <!----><i class="el-icon-check"></i><span>
          Accept
        </span></button> <button type="button" class="el-button el-button--default el-button--mini">
                <!----><i class="el-icon-close"></i><span>
          Decline
        </span></button></td>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/project/CC" class="">
                  Project C
                </a>
                <mock-el-tooltip content="Project C has a pending membership request." placement="right"><span class="notification"></span></mock-el-tooltip>
              </div>
            </td>
            <td>
              Requesting access
            </td>
            <td>
              <!----> <span>0</span></td>
            <td>
              <!---->
              <!---->
              <!---->
              <!---->
            </td>
          </tr>
          <tr>
            <td>
              <div class="sm-table-cell"><a href="/project/DD" class="">
                  Project D
                </a>
                <!---->
              </div>
            </td>
            <td>
              Project manager
            </td>
            <td><a href="/datasets?prj=DD&amp;subm=22333" class="">
                20
              </a>
              <!---->
            </td>
            <td>
              <!----> <button disabled="disabled" type="button" class="el-button el-button--default el-button--mini is-disabled">
                <!----><i class="el-icon-arrow-right"></i><span>
          Leave
        </span></button>
              <!---->
              <!---->
            </td>
          </tr>
        </table>
        <div class="el-row"><button type="button" class="el-button el-button--default" style="float: right; margin: 10px 0px;">
            <!---->
            <!----><span>
      Create project
    </span></button></div>
      </div>
    </div>
    <section>
      <h2>API access</h2>
      <div class="action-with-message">
        <p class="max-w-measure-4">
          To <a href="https://github.com/metaspace2020/metaspace/tree/master/metaspace/python-client">access METASPACE programmatically</a>
          or integrate with trusted third-party applications, an API key can be used to avoid sharing your password.
        </p>
        <!---->
      </div>
      <div class="el-row">
        <div class="action-with-message">
          <div class="max-w-measure-1 el-input el-input-group el-input-group--append">
            <!----><input type="password" readonly="readonly" autocomplete="off" class="el-input__inner">
            <!---->
            <!---->
            <div class="el-input-group__append">
              <mock-el-tooltip manual="true" content="Copied!" placement="right"><button type="button" class="el-button el-button--default" title="Copy to clipboard">
                  <!----><i class="el-icon-document-copy"></i>
                  <!----></button></mock-el-tooltip>
            </div>
            <!---->
          </div> <button type="button" class="el-button el-button--danger">
            <!---->
            <!----><span>
            Revoke key
          </span></button>
        </div>
      </div>
    </section>
    <section>
      <h2>Delete account</h2>
      <div class="action-with-message">
        <p class="max-w-measure-3">
          If you delete your METASPACE account, you can optionally remove all of your datasets.
          If datasets are not removed, the private data will still be accessible by the group members only.
        </p> <button type="button" class="el-button el-button--danger" title="Delete account">
          <!---->
          <!----><span>
          Delete account
        </span></button>
      </div>
    </section>
  </div>
</div>
`;
